### PR TITLE
Fixes extraction of nested expressions with more than two filter

### PR DESCRIPTION
**Pull request by mholthausen**
This MR

- fixes an error by extraction expressions with more than two filter expressions.  
With the previous behavior, the operators to connect the filter and the filters themself were not correctly converted to the `geostyler-style` when there were more than two filters in an expression
- provides an updated `package-lock.json`

@jansule @KaiVolland please review

### CODE REVIEW

File: src/MapfileStyleParser.ts [hunk 1]
```diff
@@ -979,11 +982,7 @@ export class MapfileStyleParser implements StyleParser {
       case ')':
         parantheseCount--;
         if (parantheseCount === 0) {
-          const operators: string = nestedExpressions[0]
-            ? mapfileExpression.substring(nestedExpressions[0].length, fromIndex)
-            : mapfileExpression.substring(0, fromIndex);
-          const negationOperator: string = operators.replace(/\s*(AND|&&|OR|\|\|)\s*/, '');
-          nestedExpressions.push(negationOperator + mapfileExpression.substring(fromIndex, index + 1));
+          nestedExpressions.push(mapfileExpression.substring(fromIndex, index + 1));
```
Comment (KaiVolland) [inline src/MapfileStyleParser.ts:985]: Is the operator not needed anymore?

Comment (mholthausen) [reply src/MapfileStyleParser.ts:985]: The `negationOperator` led to an incorrect interpretation or composition of the defined filters.
For example 

`EXPRESSION (('[class]' = 'minorroads') AND ('[tracktype]' != 'grade1') AND ('[type]' = 'track'))`

resulted in

```javscript
"&&",
["==", "class", "minorroads"],
["!=", "tracktype", "grade1"],
["!=", "tracktype", "grade1"]
```

Removing the `negationOperator` resulted in the correct interpretation:

```javscript
"&&",
["==", "class", "minorroads"],
["!=", "tracktype", "grade1"],
["==", "type", "track"]
```

Comment (KaiVolland) [reply src/MapfileStyleParser.ts:985]: Thx for the explanation.

**Review by KaiVolland — APPROVED** [review]

**Comment by mholthausen** [comment]
I have added a small test (`can read a simple MapFile with several filter expressions`) to observe the behavior in case of possible future changes

**Review by jansule — APPROVED** [review]

**Pull request merged**